index.js

```diff
@@ -581,15 +581,17 @@ var torrentStream = function(link, opts, cb) {
 		if (engine.bitfield) wire.bitfield(engine.bitfield);
 	});
 
-	swarm.pause();
+	var pauseSwarm = true;
 
 	if (link.files && engine.metadata) {
 		swarm.resume();
+		pauseSwarm = false;
 		ontorrent(link);
 	} else {
 		fs.readFile(torrentPath, function(_, buf) {
 			if (destroyed) return;
 			swarm.resume();
+			pauseSwarm = false;
```

Comment:
Fixing wire reconnection checks for a very particular case in which someone resumes a paused swarm. By using ``swarm.pause()`` right before ``swarm.resume()`` at the very start of ``torrent-stream`` this behavior is reproduce for no reason, and it will trigger wire reconnection.

See: https://github.com/jaruba/peer-wire-swarm/blob/master/index.js#L177